homeassistant/components/risco/__init__.py

```diff
@@ -0,0 +1,49 @@
+"""The Risco integration."""
+import asyncio
+
+from pyrisco import RiscoAPI
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_PIN, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+
+PLATFORMS = ["alarm_control_panel"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Risco component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Risco from a config entry."""
+    data = entry.data
+    risco = RiscoAPI(data[CONF_USERNAME], data[CONF_PASSWORD], data[CONF_PIN])
+    await risco.login()
```

Comment:
Does this throw an error when login fails?

Or the connection cannot be made? If so, it could be caught and raise a `ConfigEntryNotReady`, so Home Assistant will try again later.

Reply:
Good point. I think it's important to differentiate between something like a network failure, and an authentication an error. In the former case, it makes sense to retry, but with the latter, there's probably not much point in retrying. I'll add some granularity there.